openpype/hosts/houdini/api/lib.py

```
@@ -989,22 +1001,30 @@ def self_publish():
     if result:
         return
 
-    current_node = hou.node(".")
-    inputs_paths = find_rop_input_dependencies(
-        current_node.inputDependencies()
-    )
-    inputs_paths.append(current_node.path())
+    if not nodes:
+        nodes = [hou.node(".")]
+
+    inputs_paths = [node.path() for node in nodes]
 
     host = registered_host()
-    context = CreateContext(host, reset=True)
+    context = CreateContext(host)
 
-    for instance in context.instances:
-        node_path = instance.data.get("instance_node")
-        instance["active"] = node_path and node_path in inputs_paths
+    if mode == 0:
+        for instance in context.instances:
+            node_path = instance.data.get("instance_node")
+            instance["active"] = node_path and node_path in inputs_paths
```

Should this have a `else: raise ValueError("Unsupported mode")` fallback?